### ISSUE TITLE
[a11y]: Issue in DataTable at 400% zoom

### ISSUE THREAD

**Issue by Vinodbs** [carbon-design-system/carbon#16199]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

v1.50.0

### React version

v17.0.2

### Automated testing tool and ruleset

Manual

### Description

When the browser zoom is used to scale content to 400%, the DataTable component should reflow- i.e., it should be presented in one column so that scrolling in more than one direction is not necessary.

### Reproduction/example

https://stackblitz.com/edit/github-mbbyug?file=src%2FApp.jsx

### Steps to reproduce

On zoom , the content of the table reflow ( be presented in one column so that scrolling in more than one direction is not necessary) is not happening

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by laurenmrice** [comment]
@Vinodbs The Data table is reflowing as expected at 400% zoom and we don't want to narrow the view down to just one column especially if the data table variant is more complex like with expansion, etc.
